File: python-puka/new_task.py
#!/usr/bin/env python
import puka
import sys

client = puka.Client("amqp://localhost/")
promise = client.connect()
client.wait(promise)


promise = client.queue_declare(queue='task_queue', durable=True)
client.wait(promise)

message = ' '.join(sys.argv[1:]) or "Hello World!"
promise = client.basic_publish(exchange='',
                               routing_key='task_queue',
                               body=message,
                               headers={'delivery_mode': 2})
client.wait(promise)
print " [x] Sent %r" % (message,)

client.close()

File: javascript-nodejs/src/new_task.js
#!/usr/bin/env node

const amqp = require('amqplib');

async function main() {
    const connection = await amqp.connect(process.env.AMQP_URL || 'amqp://localhost');
    const channel = await connection.createChannel();

    const queue = 'task_queue';
    const msg = process.argv.slice(2).join(' ') || "Hello World!";

    await channel.assertQueue(queue, {
        durable: true,
        arguments: { 'x-queue-type': 'quorum' }
    });
    channel.sendToQueue(queue, Buffer.from(msg), {
        persistent: true
    });
    console.log(" [x] Sent '%s'", msg);

    setTimeout(function() {
        connection.close();
        process.exit(0);
    }, 500);
}

main();
